mlir/lib/Dialect/Arith/Utils/Utils.cpp

```diff
@@ -100,12 +100,20 @@ llvm::SmallBitVector mlir::getPositionsOfShapeOne(unsigned rank,
   return dimsToProject;
 }
 
+Value mlir::getValueOrCreateConstantIntOp(OpBuilder &b, Location loc,
+                                          Type targetType, OpFoldResult ofr) {
+  if (auto value = dyn_cast_if_present<Value>(ofr))
+    return value;
+  auto attr = cast<IntegerAttr>(dyn_cast_if_present<Attribute>(ofr));
```

Comment:
What if `dyn_cast_if_present` returns null? Should we just have a separate null check upfront?

Reply:
ofr can be either Value or Attribute, why would it fail?